File: HotDesk_PRAI/Backup/Library/Library.Root/TemplateField/deletefield.vb
Imports System.Web.UI.WebControls

Public Class deletefield
    Implements System.Web.UI.ITemplate

    Public Const LabelHeaderID As String = "lbldel"
    Public Const LiteralItemID As String = "ltritem"
    Private plittype As ListItemType

    Public Sub New(ByVal type As ListItemType)
        MyBase.New()
        Me.plittype = type
    End Sub

    Public Sub InstantiateIn(ByVal container As System.Web.UI.Control) Implements System.Web.UI.ITemplate.InstantiateIn
        Dim ltritem As Literal
        Dim lbdel As Label

        Select Case Me.plittype
            Case listItemType.Header
                lbdel = New Label()
                lbdel.Text = "Del."
                lbdel.ID = "lbldel"
                lbdel.EnableViewState = False
                container.Controls.Add(lbdel)
            Case ListItemType.Item
                ltritem = New Literal()
                ltritem.ID = "ltritem"
                ltritem.EnableViewState = False
                container.Controls.Add(ltritem)
        End Select
    End Sub

End Class

File: HotDesk_PRAI/Library/Library.Root/TemplateField/deletefield.cs
using System.Web.UI.WebControls;

public class deletefield : System.Web.UI.ITemplate
{
    public const string LabelHeaderID = "lbldel";
    public const string LiteralItemID = "ltritem";
    private ListItemType plittype;

    public deletefield(ListItemType type)
    {
        this.plittype = type;
    }

    public void InstantiateIn(System.Web.UI.Control container)
    {
        Literal ltritem;
        Label lbdel;

        switch (this.plittype)
        {
            case ListItemType.Header:
                lbdel = new Label();
                lbdel.Text = "Del.";
                lbdel.ID = "lbldel";
                lbdel.EnableViewState = false;
                container.Controls.Add(lbdel);
                break;
            case ListItemType.Item:
                ltritem = new Literal();
                ltritem.ID = "ltritem";
                ltritem.EnableViewState = false;
                container.Controls.Add(ltritem);
                break;
        }
    }
}
